pkg/operatorsource/configuring.go

```diff
@@ -95,20 +132,86 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	}
 
 	// If we are here, the given CatalogSourceConfig object already exists.
-	cscNamespacedName := types.NamespacedName{Name: in.Name, Namespace: in.Namespace}
-	cscExisting := marketplace.CatalogSourceConfig{}
-	err = r.client.Get(ctx, cscNamespacedName, &cscExisting)
+	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
 	if err != nil {
 		r.logger.Errorf("Unexpected error while getting CatalogSourceConfig: %s", err.Error())
 		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
 		return
 	}
 
+	r.logger.Info("CatalogSourceConfig object has been updated successfully")
+
+	nextPhase = phase.GetNext(phase.Succeeded)
+	return
+}
+
+// downloadManifests abstracts away the download of package metadata from the operatorsource endpoint
```

Comment:
- You are not downloading the manifests here.
- Let us be consistent and use `OperatorSource`.
- You don't need to say `abstract away`. Rather state what the function is doing.